### ISSUE TITLE
[BUG] symbol selector doesn't appear to work in the rbl module

### ISSUE THREAD

**Issue by smfreegard** [rspamd/rspamd#4358]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Tried ASAN package and obtained the ASAN report (if possible): https://rspamd.com/doc/faq.html#asan-builds 
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**

Using `selector = "symbol('BITCOIN_ADDR'):options.digest('hex','sha1')"` doesn't appear to work in the rbl module.   The symbol is raised, but the lookup doesn't run at all.

With `selectors` added to debug, I see `selectors; init.lua:90: no value extracted for symbol`, so I suspect it might be due to the lookup being run before the symbol is available.

### Steps to Reproduce

I created a rbl rule like the following:

`````
    abusix_dnsbls_btc {
        symbol = "RBL_AMI_BTC";
        rbl = "<KEY>.btc-wallets.mail-beta.abusix.zone";
        selector = "symbol('BITCOIN_ADDR'):options.digest('hex','sha1')";
    }
`````

**Expected behavior**

RBL request to be sent containing the BTC wallet address as a SHA1 value if the BITCOIN_ADDR symbol is raised.

### Versions

rspamd 3.2 on Debian 11

**Comment by vstakhov** [comment]
Yes, it is an annoying issue as selectors are parsed during the configuration stage, so they clearly can add a dependency. The problem is that a selector is not aware of the caller's symbol, so it is quite tricky to pass one throughout the API.

**Comment by leahoswald** [comment]
Same here for the settings config and the ratelimit module. It looks like the symbol selector doesn't work in most (maybe all) all because the symbols are missing.

**Comment by fatalbanana** [comment]
Applying the undocumented `require_symbols` setting to the rule should help:

~~~hcl
require_symbols = ["BITCOIN_ADDR"];
~~~

**Comment by leahoswald** [comment]
An undocumented setting... I love it! Not

But in which modules is this available? And what exactly does it?
